migrations/dynamodb-migrations/2021-08-04T10:06:50Z-migrate-sampleids-to-experiment.py

```diff
@@ -0,0 +1,156 @@
+import boto3
+from datetime import datetime
+import hashlib
+import json
+from collections import OrderedDict
+
+### Background
+# This is a migration for PR 408 - Fix sample order moves storage of samples from projects to experiments.
+
+client = boto3.client('dynamodb')
+
+environment = 'production'
+
+experiments_table = f'experiments-{environment}'
+projects_table = f'projects-{environment}'
+samples_table = f'samples-{environment}'
+
+experiments = client.scan(
+    TableName=experiments_table,
+    ProjectionExpression="experimentId, createdAt, projectId",
+)['Items']
+
+## >> Uncomment lines below to test with a single experiment
+# test_experiment_id = "b947598cc38860660a82ffd508e2b068"
+
+# experiments = client.get_item(
+#     TableName=experiments_table,
+#     Key={"experimentId": { "S" : test_experiment_id }},
+#     ProjectionExpression="experimentId, createdAt, projectId"
+# )['Item']
+# experiments = [experiments]
+
+## << Uncomment lines above to test
+
+# function to create gem2s hash
+def create_gem2s_hash(experiment, project, samples):
+
+    organism = None # default value
+
+    if experiment['meta']['M']['organism'].get('S'):
+        organism = experiment['meta']['M']['organism']['S']
+
+    # Filter 'ids' key which is present in older samples object
+    sample_ids = [sample_id for sample_id in samples['M'] if sample_id != 'ids']
+
+    sample_names = []
+
+    # Sample names
+    for sample_id in sample_ids:
+        sample_names.append(samples['M'][sample_id]['M']['name']['S'])
+
+    # Sample IDS and names are sorted to be invariant
+    sample_ids.sort()
+    sample_names.sort()
+
+    input_type = experiment["meta"]['M']["type"].get('S', '10x')
```

Comment:
this is needed in case some experiment was uploaded with an incorrect format and then the input type was not set. Shouldn't be needed.